platform/binance/transaction.go

```diff
@@ -11,180 +12,111 @@ import (
-	// Verify if the tx has more them one asset
-	if srcTx.HasChildren == 1 {
-		txs := make(blockatlas.TxPage, 0)
-		// Parse all assets as a transaction
-		for _, subTx := range srcTx.SubTxsDto.SubTxDtoList.getTxs() {
-			// If this is not called from a block observer_test, only get the user txs/assets
-			if !subTx.containAddress(address) {
-				continue
-			}
-			// Recursive call to normalize the tx
-			newTxs, ok := normalizeTransfer(tx, subTx, token, address)
-			if !ok {
-				continue
-			}
-			txs = append(txs, newTxs...)
-		}
-		if len(txs) == 0 {
-			return txs, false
+// Converts multiple transactions
+func NormalizeTxs(srcTxs []Tx, address string) (txs []blockatlas.Tx) {
+	for _, srcTx := range srcTxs {
+		tx, ok := NormalizeTx(srcTx, address)
+		if !ok {
+			continue
 		}
-		return txs, true
+		txs = append(txs, tx...)
 	}
+	return
+}
 
-	// Verify if this is the same asset we are looking for
-	if len(token) > 0 && srcTx.Asset != token {
-		return blockatlas.TxPage{tx}, false
+// Converts a Binance transaction into the generic model
+func NormalizeTx(t Tx, address string) (blockatlas.TxPage, bool) {
+	rawTx := blockatlas.Tx{
+		ID:       t.TxHash,
+		Coin:     coin.BNB,
+		From:     t.FromAddr,
+		To:       t.ToAddr,
+		Fee:      blockatlas.Amount(t.getFee()),
+		Date:     t.blockTimestamp(),
+		Block:    t.BlockHeight,
+		Status:   t.getStatus(),
+		Error:    t.getError(),
+		Sequence: t.Sequence,
+		Memo:     t.Memo,
+	}
+	rawTx.Direction = rawTx.GetTransactionDirection(address)
+
+	switch t.Type {
+	case TxTransfer:
+		normalized, ok := normalizeTransfer(rawTx, t)
+		if !ok {
+			return blockatlas.TxPage{}, false
+		}
+		return blockatlas.TxPage{normalized}, true
 	}
 
-	tx.From = srcTx.FromAddr
-	tx.To = srcTx.ToAddr
+	return blockatlas.TxPage{}, false
+}
 
+func normalizeTransfer(t blockatlas.Tx, srcTx Tx) (blockatlas.Tx, bool) {
 	bnbCoin := coin.Coins[coin.BNB]
-	value := numbers.DecimalExp(string(srcTx.Value), 8)
+	value := blockatlas.Amount(numbers.DecimalExp(srcTx.Value, 8))
+
+	// Condition for native transfer e.g.: BNB
 	if srcTx.Asset == bnbCoin.Symbol {
-		// Condition for native transfer (BNB)
-		tx.Meta = blockatlas.Transfer{
-			Value:    blockatlas.Amount(value),
-			Symbol:   bnbCoin.Symbol,
+		t.Type = blockatlas.TxTransfer
+		t.Meta = blockatlas.Transfer{
 			Decimals: bnbCoin.Decimals,
+			Symbol:   bnbCoin.Symbol,
+			Value:    value,
 		}
-		return blockatlas.TxPage{tx}, true
+		return t, true
 	}
 
-	// Condition for native token transfer
-	tx.Meta = blockatlas.NativeTokenTransfer{
-		TokenID:  srcTx.Asset,
-		Symbol:   TokenSymbol(srcTx.Asset),
-		Value:    blockatlas.Amount(value),
+	// Condition for BEP2 token transfer e.g.: TWT-8C2
+	t.Type = blockatlas.TxNativeTokenTransfer
+	t.Meta = blockatlas.NativeTokenTransfer{
 		Decimals: bnbCoin.Decimals,
 		From:     srcTx.FromAddr,
+		Symbol:   TokenSymbol(srcTx.Asset),
 		To:       srcTx.ToAddr,
+		TokenID:  srcTx.Asset,
+		Value:    value,
+		Name:     "",
```

Comment:
if I am not mistaken, this can be skipped (the default value is "")

Reply:
Yes and know, ideally we need to return token name for as far is it defined on struct. For now removed.